sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -1070,15 +1070,32 @@ public static RetryConfiguration create(
      *
      * <pre>{@code
      * PCollection<Void> firstWriteResults = data.apply(JdbcIO.write()
-     *     .withDataSourceConfiguration(CONF_DB_1).withResults());
+     *     .withDataSourceConfiguration(CONF_DB_1).withVoidResults());
      * data.apply(Wait.on(firstWriteResults))
      *     .apply(JdbcIO.write().withDataSourceConfiguration(CONF_DB_2));
      * }</pre>
      */
-    public WriteVoid<T> withResults() {
+    public WriteVoid<T> withVoidResults() {
       return inner;
     }
 
+    /**
+     * Returns {@link WriteWithResults} transform that could return a specific result.
+     *
+     * <p>See {@link WriteWithResults}
+     */
+    public <V> WriteWithResults<T, V> withReturningResults(RowMapper<V> rowMapper) {
```

Comment:
How about renaming it the following way?

- `withVoidResults` -> `withoutResults`
- `withReturningResults` -> `withResults`

I suppose this way should be more concise, what do you think?

Reply:
It is a good question. I thought about this: `withVoidResults` -> `withoutResults`. But initially, this method was named `withResults`, because we can wait on such results even if they are Void.

Reply:
I'd suggest to keep and deprecate the old `withResults()` and name this method (`withReturningResults()`)  as `withWriteResults()` like BigtableIO [does](https://github.com/apache/beam/blob/f881a412fe85c64b1caf075160a6c0312995ea45/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java#L715) (to keep the same naming policy for the similar functionality across different IOs).